packages/js/product-editor/src/components/iframe-editor/header-toolbar/header-toolbar.tsx

```diff
@@ -87,42 +109,93 @@ export function HeaderToolbar( {
 		}
 	}, [ isInserterOpened, setIsInserterOpened ] );
 
+	useEffect( () => {
+		// If we have a new block selection, show the block tools
+		if ( hasBlockSelection ) {
+			setIsBlockToolsCollapsed( false );
+		}
+	}, [ hasBlockSelection ] );
+
+	const renderBlockToolbar =
+		isWpVersion( '6.5', '>=' ) || getGutenbergVersion() > 17.3;
+
 	return (
 		<NavigableToolbar
 			className="woocommerce-iframe-editor__header-toolbar"
 			aria-label={ toolbarAriaLabel }
 		>
 			<div className="woocommerce-iframe-editor__header-toolbar-left">
-				<ToolbarItem
-					ref={ inserterButton }
-					as={ Button }
-					className="woocommerce-iframe-editor__header-toolbar-inserter-toggle"
-					variant="primary"
-					isPressed={ isInserterOpened }
-					onMouseDown={ (
-						event: MouseEvent< HTMLButtonElement >
-					) => {
-						event.preventDefault();
-					} }
-					onClick={ toggleInserter }
-					disabled={ ! isInserterEnabled }
-					icon={ plus }
-					label={
-						! isInserterOpened
-							? __( 'Add', 'woocommerce' )
-							: __( 'Close', 'woocommerce' )
-					}
-					showTooltip
-				/>
-				{ isLargeViewport && (
+				<div className="woocommerce-iframe-editor-document-tools">
 					<ToolbarItem
-						as={ ToolSelector }
-						disabled={ isTextModeEnabled }
+						ref={ inserterButton }
+						as={ Button }
+						className="woocommerce-iframe-editor__header-toolbar-inserter-toggle"
+						variant="primary"
+						isPressed={ isInserterOpened }
+						onMouseDown={ (
+							event: MouseEvent< HTMLButtonElement >
+						) => {
+							event.preventDefault();
+						} }
+						onClick={ toggleInserter }
+						disabled={ ! isInserterEnabled }
+						icon={ plus }
+						label={
+							! isInserterOpened
+								? __( 'Add', 'woocommerce' )
+								: __( 'Close', 'woocommerce' )
+						}
+						showTooltip
 					/>
+					{ isLargeViewport && (
+						<ToolbarItem
+							as={ ToolSelector }
+							disabled={ isTextModeEnabled }
+						/>
+					) }
+					<ToolbarItem as={ EditorHistoryUndo } />
+					<ToolbarItem as={ EditorHistoryRedo } />
+					<ToolbarItem as={ DocumentOverview } />
+				</div>
+				{ hasFixedToolbar && isLargeViewport && renderBlockToolbar && (
+					<>
+						<div
+							className={ classnames(
+								'selected-block-tools-wrapper',
+								{
+									'is-collapsed': isBlockToolsCollapsed,
+								}
+							) }
+						>
+							{ /* @ts-expect-error missing type */ }
+							<BlockToolbar hideDragHandle />
```

Comment:
This component was added new, and allows us to render the block toolbar anywhere, in this case we could render it inline within the header to match the other editors.